warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/balancer/ShardRendezvousHostBalancer.java

```diff
@@ -159,10 +164,17 @@ protected Function<String,List<TabletServerId>> getServerPartitioner(Collection<
             throw new IllegalStateException("Tier configuration is not set");
         }
 
+        final HashSet<Pattern> unMatchedPatterns = new HashSet<>();
+        unMatchedPatterns.addAll(configuredTiers.stream().map(Map.Entry::getValue).map(Matcher::pattern).collect(Collectors.toSet()));
+
         TreeMap<Long,List<TabletServerId>> serverPartitioningMap = new TreeMap<>();
 
+        // Ensure each tier has a list of tservers. This will ensure that when a tiers regex matches nothing that the tier has an empty list. This will cause
+        // its tablets to not be assigned anywhere.
+        configuredTiers.forEach(e -> serverPartitioningMap.put(e.getKey(), new ArrayList<>()));
```

Comment:
Adding this line avoids sending tablets in a tier to a neighboring tier when the tier has no tservers.